File: testled.py
#!/usr/bin/python

# NeoPixel library strandtest example
# Author: Tony DiCola (tony@tonydicola.com)
#
# Direct port of the Arduino NeoPixel library strandtest example.  Showcases
# various animations on a strip of NeoPixels.
import time

from neopixel import *

import argparse
import signal
import sys

def signal_handler(signal, frame):
        colorWipe(strip, Color(0,0,0))
        sys.exit(0)

# LED strip configuration:
LED_COUNT      = 2       # Number of LED pixels.
LED_PIN        = 18      # GPIO pin connected to the pixels (18 uses PWM!).
#LED_PIN        = 10      # GPIO pin connected to the pixels (10 uses SPI /dev/spidev0.0).
LED_FREQ_HZ    = 800000  # LED signal frequency in hertz (usually 800khz)
LED_DMA        = 5       # DMA channel to use for generating signal (try 5)
LED_BRIGHTNESS = 255     # Set to 0 for darkest and 255 for brightest
LED_INVERT     = False   # True to invert the signal (when using NPN transistor level shift)
LED_CHANNEL    = 0       # set to '1' for GPIOs 13, 19, 41, 45 or 53
LED_STRIP      = ws.WS2811_STRIP_GRB   # Strip type and colour ordering



# Define functions which animate LEDs in various ways.
def colorWipe(strip, color, wait_ms=50):
	"""Wipe color across display a pixel at a time."""
	for i in range(strip.numPixels()):
		strip.setPixelColor(i, color)
		strip.show()
		time.sleep(wait_ms/1000.0)

# Main program logic follows:
if __name__ == '__main__':
        signal.signal(signal.SIGINT, signal_handler)

	# Create NeoPixel object with appropriate configuration.
	strip = Adafruit_NeoPixel(LED_COUNT, LED_PIN, LED_FREQ_HZ, LED_DMA, LED_INVERT, LED_BRIGHTNESS, LED_CHANNEL, LED_STRIP)

	# Intialize the library (must be called once before other functions).
	strip.begin()

	print ('Press Ctrl-C to quit.')
	while True:
		print ('Color Red')
		colorWipe(strip, Color(255, 0, 0))  # Red wipe
		time.sleep(2)
		print ('Color Green')
		colorWipe(strip, Color(0, 255, 0))  # Green wipe
		time.sleep(2)
		print ('Color Blue')
		colorWipe(strip, Color(0, 0, 255))  # Blue wipe
		time.sleep(2)

File: testled.js
#!/usr/bin/env node

var ws281x = require('rpi-ws281x-native');
var NUM_LEDS =  2
var r = 255
var g = 0
var b = 0
pixelData = new Uint32Array(NUM_LEDS);
ws281x.init(NUM_LEDS);

// trap the SIGINT and reset before exit
process.on('SIGINT', function () {
  ws281x.reset();
  process.nextTick(function () { process.exit(0); });
});


//  animation-loop
var offset = 0;

setInterval(function () {
  if (r==255) console.log('Red');
  if (g==255) console.log('Green');
  if (b==255) console.log('Blue');

  for (var i = 0; i < NUM_LEDS; i++) {
    pixelData[i] = rgb2Int(r,g,b);
  }
  ws281x.render(pixelData);

  if (r==255) {
    g=255; r=0;
  } else if (g==255){
    b=255; g=0;
  } else {
    r=255; b=0;
  }
}, 2000 );

function rgb2Int(r, g, b) {
  return ((r & 0xff) << 16) + ((g & 0xff) << 8) + (b & 0xff);
}

console.log('Press <ctrl>+C to exit.');


